File: cli/src/agent/templates.ts
/**
 * Pre-built agent configuration templates.
 */

import type { AgentConfig } from './types.js';

export const templates: Record<string, Omit<AgentConfig, 'slug'>> = {
  'seo-agent': {
    name: 'SEO Analyst',
    expertise: ['seo', 'content-analysis', 'keyword-research'],
    personality: 'Analytical and data-driven. Focuses on search visibility.',
    heartbeatCron: 'daily',
    tools: ['nex_search', 'nex_ask', 'nex_remember', 'nex_record_list'],
  },
  'lead-gen': {
    name: 'Lead Generator',
    expertise: ['prospecting', 'enrichment', 'outreach'],
    personality: 'Proactive hunter. Identifies and qualifies leads.',
    heartbeatCron: '0 */6 * * *',
    tools: ['nex_search', 'nex_record_list', 'nex_record_create', 'nex_remember'],
  },
  'enrichment': {
    name: 'Data Enricher',
    expertise: ['data-enrichment', 'research', 'validation'],
    personality: 'Thorough researcher. Fills in missing data.',
    heartbeatCron: '0 */4 * * *',
    tools: ['nex_search', 'nex_ask', 'nex_record_get', 'nex_record_update', 'nex_remember'],
  },
  'research': {
    name: 'Research Analyst',
    expertise: ['market-research', 'competitive-analysis', 'trend-analysis'],
    personality: 'Curious and systematic. Connects dots across data.',
    heartbeatCron: 'daily',
    tools: ['nex_search', 'nex_ask', 'nex_remember'],
  },
  'customer-success': {
    name: 'Customer Success',
    expertise: ['relationship-management', 'health-scoring', 'renewal-tracking'],
    personality: 'Empathetic and proactive. Anticipates customer needs.',
    heartbeatCron: '0 */8 * * *',
    tools: ['nex_search', 'nex_ask', 'nex_record_list', 'nex_record_get', 'nex_remember'],
  },
  'team-lead': {
    name: 'Team Lead',
    expertise: ['general', 'research', 'analysis', 'communication', 'planning', 'orchestration'],
    personality: 'You are the Team Lead — the primary interface between the human and the specialist agents. You understand objectives, break them into tasks, delegate to the right specialists, and synthesize results. When the user says what they want, you figure out the path to get there using the context graph and available agents. You always respond first, then coordinate behind the scenes.',
    heartbeatCron: 'manual',
    tools: ['nex_search', 'nex_ask', 'nex_remember', 'nex_record_list', 'nex_record_get', 'nex_record_create', 'nex_record_update'],
  },
  'founding-agent': {
    name: 'Team Lead',
    expertise: ['general', 'research', 'analysis', 'communication', 'planning', 'orchestration'],
    personality: 'Versatile and proactive. Your first AI team member — handles everything from research to outreach until specialized agents are added. Delegates to specialists when they exist.',
    heartbeatCron: 'daily',
    tools: ['nex_search', 'nex_ask', 'nex_remember', 'nex_record_list', 'nex_record_get', 'nex_record_create', 'nex_record_update'],
  },
};

File: internal/agent/templates.go
package agent

// Templates is the built-in agent template registry (keyed by slug, without slug field set).
var Templates = map[string]AgentConfig{
	"seo-agent": {
		Name:          "SEO Analyst",
		Expertise:     []string{"seo", "content-analysis", "keyword-research"},
		Personality:   "Analytical and data-driven...",
		HeartbeatCron: "daily",
		Tools:         []string{"nex_search", "nex_ask", "nex_remember", "nex_record_list"},
	},
	"lead-gen": {
		Name:          "Lead Generator",
		Expertise:     []string{"prospecting", "enrichment", "outreach"},
		Personality:   "Proactive hunter...",
		HeartbeatCron: "0 */6 * * *",
		Tools:         []string{"nex_search", "nex_record_list", "nex_record_create", "nex_remember"},
	},
	"enrichment": {
		Name:          "Data Enricher",
		Expertise:     []string{"data-enrichment", "research", "validation"},
		Personality:   "Thorough researcher...",
		HeartbeatCron: "0 */4 * * *",
		Tools:         []string{"nex_search", "nex_ask", "nex_record_get", "nex_record_update", "nex_remember"},
	},
	"research": {
		Name:          "Research Analyst",
		Expertise:     []string{"market-research", "competitive-analysis", "trend-analysis"},
		Personality:   "Curious and systematic...",
		HeartbeatCron: "daily",
		Tools:         []string{"nex_search", "nex_ask", "nex_remember"},
	},
	"customer-success": {
		Name:          "Customer Success",
		Expertise:     []string{"relationship-management", "health-scoring", "renewal-tracking"},
		Personality:   "Empathetic and proactive...",
		HeartbeatCron: "0 */8 * * *",
		Tools:         []string{"nex_search", "nex_ask", "nex_record_list", "nex_record_get", "nex_remember"},
	},
	"team-lead": {
		Name:          "Team Lead",
		Expertise:     []string{"general", "research", "analysis", "communication", "planning", "orchestration"},
		Personality:   "You are the Team Lead — the primary interface...",
		HeartbeatCron: "manual",
		Tools:         []string{"nex_search", "nex_ask", "nex_remember", "nex_record_list", "nex_record_get", "nex_record_create", "nex_record_update"},
	},
	"founding-agent": {
		Name:          "Team Lead",
		Expertise:     []string{"general", "research", "analysis", "communication", "planning", "orchestration"},
		Personality:   "Versatile and proactive...",
		HeartbeatCron: "daily",
		Tools:         []string{"nex_search", "nex_ask", "nex_remember", "nex_record_list", "nex_record_get", "nex_record_create", "nex_record_update"},
	},
}
